source/configure: add config in db editing limitations (#6000)
mattermost/docs: source/configure/configuation-in-mattermost-database.rst

* Ensure all servers in a High Availability deployment have the same configuration, even when new servers are added to the cluster.
* Automatically deploy SAML certificates and keys to all servers in the cluster.

> Note that once you start using configuration in the database, you can't manually edit the active configuration row. You will either need to do one of:

Reviewer: Maybe also better to clarify that one "shouldn't" do this, rather than "can't". Because an admin can very well edit some DB rows. We should also clarify why one shouldn't do this, so that the impact is clear.

Author: Makes sense, they can actually *edit* but it wouldn't take any effect.

* Ensure all servers in a High Availability deployment have the same configuration, even when new servers are added to the cluster.
* Automatically deploy SAML certificates and keys to all servers in the cluster.

> Note that once you start using configuration in the database, you shouldn't manually edit the active configuration row. You will either should do one of to edit/update the configuration:

Reviewer 2: ```suggestion
Note that once you start using configuration in the database, you shouldn't manually edit the active configuration row. You should edit or update the configuration in one of the following ways:
```

* Use the System Console to make changes to the configuration.
* Use ``mmctl`` to make changes to the configuration.
* Stop any of the running mattermost-server instances and edit the active configuration row directly in the ``Configurations`` table.

> The reason for this is that the Mattermost server keeps active configuration in memory and writes new one to the database only there is a change. This way we avoid polling the database to process changes to the configuration. Publishing the changes to the cluster handled by the application itself.

Reviewer 2: ```suggestion
The Mattermost server keeps active configuration in memory and writes new ones to the database only when there is a change. This way we avoid polling the database to process changes to the configuration. Publishing the changes to the cluster are handled by the application itself.
```
